server/internal/infrastructure/aws/file.go

```diff
@@ -0,0 +1,276 @@
+package aws
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/google/uuid"
+	"github.com/reearth/reearth-cms/server/internal/usecase/gateway"
+	"github.com/reearth/reearth-cms/server/pkg/asset"
+	"github.com/reearth/reearth-cms/server/pkg/file"
+	"github.com/reearth/reearthx/i18n"
+	"github.com/reearth/reearthx/rerror"
+	"github.com/samber/lo"
+)
+
+const (
+	s3AssetBasePath string = "assets"
+	fileSizeLimit   int64  = 10 * 1024 * 1024 * 1024 // 10GB
+)
+
+type fileRepo struct {
+	bucketName   string
+	baseURL      *url.URL
+	cacheControl string
+	s3Client     *s3.Client
+}
+
+func NewFile(bucketName, accessKeyID, secretAccessKey, region, baseURL, cacheControl string) (gateway.File, error) {
+	if bucketName == "" {
+		return nil, rerror.NewE(i18n.T("bucket name is empty"))
+	}
+
+	if secretAccessKey == "" || accessKeyID == "" {
+		return nil, rerror.NewE(i18n.T("no credentials provided"))
+	}
+
+	if region == "" {
+		return nil, rerror.NewE(i18n.T("region is empty"))
+	}
+
+	var u *url.URL
+	if baseURL == "" {
+		baseURL = fmt.Sprintf("https://%s.s3.amazonaws.com/", bucketName)
+	}
+
+	var err error
+	u, _ = url.Parse(baseURL)
+	if err != nil {
+		return nil, rerror.NewE(i18n.T("invalid base URL"))
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion(region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKeyID, secretAccessKey, "")),
```

Comment:
```suggestion
```

https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/

As much as possible, leave it to the SDK to load the configuration automatically, and support different configuration methods such as environment variables and files. It can also load the configuration automatically with zero configuration when running in an AWS environment.

Reply:
Done!